synapse/storage/databases/main/devices.py

```diff
@@ -1763,9 +1760,6 @@ def _update_remote_device_list_cache_entry_txn(
             table="device_lists_remote_extremeties",
             keyvalues={"user_id": user_id},
             values={"stream_id": stream_id},
-            # again, we can assume we are the only thread updating this user's
-            # extremity.
-            lock=False,
```

Comment:
Handled by unsafe to upsert mechanism these days: https://github.com/matrix-org/synapse/blob/c3a4780080a5bcb04132283c0f32f7452655792a/synapse/storage/database.py#L92

Reply:
ditto below